pi-manage

```diff
@@ -112,6 +113,35 @@ manager.add_command('event', event_manager)
 manager.add_command('api', api_manager)
 manager.add_command('ca', ca_manager)
 manager.add_command('audit', audit_manager)
+manager.add_command('hsm', hsm_manager)
+
+
+@hsm_manager.command
+def create_keys():
+    """
+    Create new encryption keys on the HSM. Be sure to first setup the HSM module, the PKCS11
+    module and the slot/password for the given HSM in your pi.cfg.
+    Set the variables PI_HSM_MODULE, PI_HSM_MODULE_MODULE, PI_HSM_MODULE_SLOT, PI_HSM_MODULE_PASSWORD.
+    """
+
+    hsm_module_name = app.config.get("PI_HSM_MODULE",
+                                     "privacyidea.lib.security.default.DefaultSecurityModule")
+    module_parts = hsm_module_name.split(".")
+    package_name = ".".join(module_parts[:-1])
+    class_name = module_parts[-1]
+    mod = __import__(package_name, globals(), locals(), [class_name])
+    hsm_class = getattr(mod, class_name)
+    hsm_parameters = {}
+    for key in app.config.keys():
+        if key.startswith("PI_HSM_MODULE_"):
+            param = key[len("PI_HSM_MODULE_"):].lower()
+            hsm_parameters[param] = app.config.get(key)
+    hsm_object = hsm_class(hsm_parameters)
```

Comment:
Couldn't we just import ``init_hsm`` from ``privacyidea.lib.crypto`` and replace all the code above with ``hsm_object = init_hsm()``?

Reply:
I already checked this. We rould run outside of an application context, since ``init_hsm`` uses ``current_app``!

So: No.

Reply:
Hm, I thought this could work because flask-script would push an application context before handling the command :-/
Anyway, I'm not so happy about the code duplication with ``init_hsm`` -- especially because this is quite complex code and it might happen that we change it at one place, but forget to change it at the other place. Also, we might add more ``hsm`` commands in the future.
Maybe we can split ``init_hsm`` into two functions, ``import_hsm(config)`` (which works without an application context) and ``init_hsm()`` which needs the application context and invokes ``import_hsm``?

Reply:
Will do so. Can you please mark this as required change in the review?